static/app/utils/replays/replayReader.tsx

```diff
@@ -198,6 +215,73 @@ export default class ReplayReader {
   private _sortedRRWebEvents: RecordingFrame[] = [];
   private _sortedSpanFrames: SpanFrame[] = [];
 
+  private _startOffsetMs = 0;
+
+  private _applyClipWindow = (clipWindow: ClipWindow) => {
+    const startedAtMs = clamp(
+      clipWindow.startTimestamp,
+      this._replayRecord.started_at.getTime(),
+      this._replayRecord.finished_at.getTime()
+    );
+    const finishedAtMs = clamp(
+      clipWindow.endTimestamp,
+      startedAtMs,
+      this._replayRecord.finished_at.getTime()
+    );
+
+    // For RRWeb frames we only trim from the end because playback will
+    // not work otherwise. The start offset is used to begin playback at
+    // the correct time.
+    this._sortedRRWebEvents = this._sortedRRWebEvents.filter(
+      frame =>
+        frame.timestamp >= this._replayRecord.started_at.getTime() &&
+        frame.timestamp <= finishedAtMs
+    );
+    this._replayRecord.finished_at = new Date(finishedAtMs);
+    this._replayRecord.duration = duration(
+      finishedAtMs - this._replayRecord.started_at.getTime()
+    );
+
+    // We also only trim from the back for breadcrumbs/spans to keep
+    // historical information about the replay, such as the current URL.
+    this._sortedBreadcrumbFrames = this._trimFramesToClipWindow(
+      this._sortedBreadcrumbFrames,
+      this._replayRecord.started_at.getTime(),
+      finishedAtMs
+    );
+    this._sortedSpanFrames = this._trimFramesToClipWindow(
+      this._sortedSpanFrames,
+      this._replayRecord.started_at.getTime(),
+      finishedAtMs
+    );
+
+    this._startOffsetMs = startedAtMs - this._replayRecord.started_at.getTime();
+  };
+
+  /**
+   * Filters out frames that are outside of the clip window, and adjusts
+   * any offset timestamps to be relative to the start offset.
+   */
+  _trimFramesToClipWindow = <T extends {timestampMs: number}>(
```

Comment:
I believe everything has an `offsetMs`, and therefore the `if` statement in the map() isn't needed.  Everything extends from `HydratedStartEndDate` or `HydratedTimestamp`

If we update the type generic it should all be safe.
```
_trimFramesToClipWindow = <T extends {timestampMs: number, offsetMs: number}>(